buildpack.toml

```diff
@@ -1,26 +1,21 @@
-api = "0.4"
+api = "0.6"
 
 [buildpack]
 id = "heroku/ruby"
-version = "1.0.2"
+version = "2.0.0"
 name = "Ruby"
 homepage = "https://github.com/heroku/buildpacks-ruby"
-ruby_version = "2.6.6"
-
-  [publish.Ignore]
-  files = [
-    "spec/"
-  ]
-
-  [[publish.Vendor]]
-  url = "https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/heroku-18/ruby-2.6.6.tgz"
-  dir = "vendor/ruby/heroku-18"
-  [[publish.Vendor]]
-  url = "https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/heroku-20/ruby-2.6.6.tgz"
-  dir = "vendor/ruby/heroku-20"
+description = "Official Heroku buildpack for using Ruby to target Cloud Native Buildpack (CNB) builds."
+keywords = ["ruby", "rails"]
 
 [[stacks]]
-id = "heroku-18"
+id = "heroku-20"
 
 [[stacks]]
-id = "heroku-20"
+id = "heroku-22"
+
+[metadata]
+ruby_url = "https://heroku-buildpack-ruby.s3.amazonaws.com/heroku-20/ruby-2.7.4.tgz"
```

Comment:
I think this `ruby_url` is now unused, since it's generated dynamically elsewhere in the buildpack?

Reply:
Fun story removing these two lines make detect not pass. This might be a bug in libcnb.rs

Reply:
The problem is the inconsistency between `RubyBuildpackMetadata` struct when it's removed. But weirdly it doesn't panic or failt to compile it just fails to pass detect ಠ_ಠ.

Reply:
The reason this happens is:
- The `buildpack.toml` file is deserialised by libcnb at runtime (not compile time), and this fails if the toml file doesn't match the provided schema
- This deserialisation in libcnb happens not only in `bin/build` but during `bin/detect` too (since the detect phase may also want metadata from the buildpack.toml, plus libcnb needs to check things like whether the buildpack API version set is a supported one etc)
- Pack CLI doesn't surface detect errors very well (eg you have to pass `--verbose` to get anything useful)

I'd love to improve the detect errors side upstream in Pack/lifecycle/the CNB spec - let's chat about this later in our 1:1 :-)

Reply:
@schneems If I fix the exit code used by libherokubuildpack:
https://github.com/heroku/libherokubuildpack/issues/54

...then I get this error message for such cases (without needing to use `--verbose`):
```
[Error: Internal Buildpack Error]
Cannot read buildpack descriptor (buildpack.toml): TOML deserialization error while reading TOML file: missing field `metadata` at line 26 column 1
```

...which is much better :-)